test/support/proposal_context.rb

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "storage"
+require "y2storage"
+
+RSpec.shared_context "proposal" do
+  using Y2Storage::Refinements::TestDevicegraph
+  using Y2Storage::Refinements::SizeCasts
+  using Y2Storage::Refinements::DevicegraphLists
+
+  before do
+    fake_scenario(scenario)
+
+    allow(Y2Storage::DiskAnalyzer).to receive(:new).and_return disk_analyzer
+    allow(disk_analyzer).to receive(:windows_partition?) do |partition|
+      !!(partition.filesystem.label =~ /indows/)
+    end
+
+    allow_any_instance_of(::Storage::Filesystem).to receive(:detect_resize_info)
+      .and_return(resize_info)
```

Comment:
Setting the resize info in the library would increase the usefulness of the tests in that the real API is used.